src/account/operations/transaction/mod.rs

```diff
@@ -178,4 +190,63 @@ impl AccountHandle {
         }
         Ok(())
     }
+
+    // Try to get a transaction confirmed and sync related account addresses when confirmed, so the outputs get
+    // available for new transactions without manually syncing (which would sync all addresses and be more heavy without
+    // extra logic)
+    fn monitor_tx_confirmation(&self, block_id: BlockId) {
+        // spawn a task which tries to get the block confirmed
+        let account = self.clone();
+        tokio::spawn(async move {
+            if let Ok(blocks) = account.client().retry_until_included(&block_id, None, None).await {
+                if let Some(confirmed_block) = blocks.first() {
+                    if confirmed_block.0 != block_id {
+                        log::debug!(
+                            "[TRANSACTION] reattached {}, new block id {}",
+                            block_id,
+                            confirmed_block.0
+                        );
+                    }
+                    if let Some(Payload::Transaction(tx_payload)) = confirmed_block.1.payload() {
+                        let TransactionEssence::Regular(regular_essence) = tx_payload.essence();
+                        if let Ok(account_addresses) = account.list_addresses().await {
+                            // Filter for addresses from the account
+                            let addresses: Vec<String> = regular_essence
+                                .outputs()
+                                .iter()
+                                .filter_map(|o| {
+                                    o.unlock_conditions()
+                                        .expect("output needs to have unlock conditions")
```

Comment:
I know this can't happen atm but it's probably more future-proof to also use an `and_then` here